File: Sorting/Insertion.java
package Sorting;
import java.util.*;

public class Insertion {
    public static void main(String args[]) {
        int i, temp, j, n;
        int arr[] = new int[10];
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the no of elements: \n");
        n = sc.nextInt();
        for(i=0; i<n; i++) {
            System.out.print("Enter value: ");
            arr[i] = sc.nextInt();
        } 
        for(i=1; i<n; i++) {
            temp = arr[i];
            j = i - 1;
            while(j>=0 && arr[j] > temp){
                arr[j+1] = arr[j];
                j--;
            }
            arr[j+1] = temp;
        }
        System.out.println("After Sorting...");
        for(i=0; i<n; i++) {
            System.out.print(arr[i] + "   ");
        }
    }
}

File: Sorting/Insertion.c
#include<stdio.h>

void main() {
    int n,i,j,a[10],temp;
    printf("Enter the number of elements: ");
    scanf("%d", &n);
    for(i=0; i<n; i++) {
        printf("Enter number: ");
        scanf("%d", &a[i]);
    }
    for(i=1; i<n; i++) {
        temp = a[i];
        j = i - 1;
        while(j>=0 && a[j] > temp){
            a[j+1] = a[j];
            j--;
        }
        a[j+1] = temp;
    }

    printf("After sorting..\n");

    for(i=0; i<n; i++) {
        printf("%d\t", a[i]);
    }
}

//Time complexity:
//Worst case O(n^2)
//Best case O(n)